### PR TITLE
Add register and token actions, improve message content and format

### PR DESCRIPTION
**Pull request by kjsanger**
Use Study rather study_id to build the message body.

Fix missing domain argument.

Bump npg-python-lib from 0.3.2 to 0.3.4

### CODE REVIEW

File: src/npg_notify/ont/event.py [hunk 1]
```diff
@@ -185,7 +192,7 @@ def body(self, *study_ids: list[str]) -> str:
                         "flowcell_id": self.flowcell_id,
                         "path": self.path,
                         "event": self.event,
-                        "studies": "\n".join([*study_ids]),
+                        "studies": "\n".join([*study_descs]),
```
Comment (mgcam) [inline src/npg_notify/ont/event.py:195]: "\n".join(study_descs) should suffice?

File: tests/ont/test_generate_email.py [hunk 2]
```diff
@@ -58,16 +59,15 @@ def test_generate_email(self):
...
-            f"The ONT run for experiment {expt}, flowcell {flowcell_id} has been {event_type}.\n"
-            "The data are available in iRODS at the following path:\n"
+        assert event.body(studies) == (
+            f"The ONT run for experiment {expt}, flowcell {flowcell_id} has been {event_type}. The data are available in iRODS at the following path:\n"
             "\n"
             f"{path}\n"
             "\n"
-            "This is an automated email from NPG. You are receiving it because you are registered\n"
-            "as a contact for one or more of the Studies listed below:\n"
+            "This is an automated email from NPG. You are receiving it because you are registered as a contact for one or more of the Studies listed below:\n"
```
Comment (mgcam) [inline tests/ont/test_generate_email.py:70]: Notifications for Illumina and PacBio platforms have a standard footer. Do we really want ONT notifications to be different?

**Review by mgcam — CHANGES_REQUESTED** [review]

File: tests/ont/test_generate_email.py (hunk 2 above)
Comment (mgcam) [reply tests/ont/test_generate_email.py:70]: Yes. Lines 152-154 were added on request by SSRs

Comment (kjsanger) [reply tests/ont/test_generate_email.py:70]: I'm also expecting a few other details in the next week or two after it's live - listing the plate barcodes and tag names/sequences.

Comment (mgcam) [reply tests/ont/test_generate_email.py:70]: That's fine.

**Review by mgcam — APPROVED** [review]

**Pull request merged**